### PR TITLE
#1918 multiple target this mappings

### PR DESCRIPTION
**Pull request by sjaakd**
I'll probably need to add more properties to make a bit more realistic

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/targetthis/ConfictsResolvedNestedMapper.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface ConfictsResolvedNestedMapper {
```
Comment (Captain1653) [inline processor/src/test/java/org/mapstruct/ap/test/targetthis/ConfictsResolvedNestedMapper.java]: Maybe javadoc @author ?

File: processor/src/test/java/org/mapstruct/ap/test/targetthis/ErroneousNestedMapper.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface ErroneousNestedMapper {
```
Comment (Captain1653) [inline processor/src/test/java/org/mapstruct/ap/test/targetthis/ErroneousNestedMapper.java]: Maybe javadoc @author?

File: processor/src/test/java/org/mapstruct/ap/test/targetthis/FlatteningMapper.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface FlatteningMapper {
```
Comment (Captain1653) [inline processor/src/test/java/org/mapstruct/ap/test/targetthis/FlatteningMapper.java]: Maybe javadoc @author?

**Comment by Captain1653** [comment]
I wanted to ask about documentation. I mean, if we allow multiply target mapping then should we add this in documentation? Am I wrong?

**Comment by sjaakd** [comment]
> I wanted to ask about documentation. I mean, if we allow multiply target mapping then should we add this in documentation? Am I wrong?

done..

File: processor/src/test/java/org/mapstruct/ap/test/targetthis/FlatteningMapper.java (hunk 3 above)
Comment (sjaakd) [reply processor/src/test/java/org/mapstruct/ap/test/targetthis/FlatteningMapper.java]: done

**Comment by sjaakd** [comment]
@filiphr , @Captain1653 .. thanks for reviewing

**Pull request merged**